src/pytest_mock/plugin.py

```diff
@@ -254,6 +258,63 @@ def dict(
                 **kwargs
             )
 
+        @overload
+        def __call__(
+            self,
+            target: str,
+            new: None = ...,
+            spec: Optional[builtins.object] = ...,
+            create: bool = ...,
+            spec_set: Optional[builtins.object] = ...,
+            autospec: Optional[builtins.object] = ...,
+            new_callable: None = ...,
+            **kwargs: Any
+        ) -> unittest.mock.MagicMock:
```

Comment:
these functions sometimes return ` unittest.mock.MagicMock:` and sometimes return `mock.MagicMock:`

Reply:
Indeed, but I don't think it matters to the type checker in the end, given that they both have the same interface, correct?

If not I don't know how to annotate that, given that the choice between mock/unittest.mock is done at runtime.

Reply:
I think it might be possible with a


```python
class UnittestMockProtocol(Protocol): ...
class MockProtocol(Protocol): ...

_MP = TypeVar("_MP", bound=Union[UnittestMockProtocol, MockProtocol]])

class MockerFixture(Generic[_MP]): ...


MockMockerFixture = MockerFixture[MockProtocol]
UnittestMockerFixture = MockerFixture[UnittestMockProtocol]
```

and then write all the methods used twice over in each of UnittestMockProtocol and MockProtocol

Reply:
otherwise codegen?

Reply:
Hmm interesting, but is it worth it? Don't `unittest.mock.MagicMock` and `mock.MagicMock` have the same interface?

Reply:
@graingert gentle ping.

Reply:
they have a different interface when you're running a more modern backport, and they are nominal types

Reply:
Indeed that can happen, but I'm not sure this will happen often in practice vs doing something more complex like codegen or explicitly declaring protocols.

We can review this however in the future if I'm wrong on that, thanks for bringing it up. 👍